src/it/java/io/weaviate/integration/SearchITest.java

```diff
@@ -0,0 +1,227 @@
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.rules.TestRule;
+
+import io.weaviate.ConcurrentTest;
+import io.weaviate.client6.WeaviateClient;
+import io.weaviate.client6.v1.collections.Property;
+import io.weaviate.client6.v1.collections.Reference;
+import io.weaviate.client6.v1.collections.VectorIndex;
+import io.weaviate.client6.v1.collections.VectorIndex.IndexingStrategy;
+import io.weaviate.client6.v1.collections.Vectorizer;
+import io.weaviate.client6.v1.collections.object.Vectors;
+import io.weaviate.client6.v1.collections.query.GroupedQueryResult;
+import io.weaviate.client6.v1.collections.query.MetadataField;
+import io.weaviate.client6.v1.collections.query.NearText;
+import io.weaviate.client6.v1.collections.query.NearVector;
+import io.weaviate.containers.Container;
+import io.weaviate.containers.Container.ContainerGroup;
+import io.weaviate.containers.Contextionary;
+import io.weaviate.containers.Weaviate;
+
+public class SearchITest extends ConcurrentTest {
+  private static final ContainerGroup compose = Container.compose(
+      Weaviate.custom()
+          .withContextionaryUrl(Contextionary.URL)
+          // .withClipInferenceApi(Multi2VecClip.URL)
+          .build(),
+      // Container.MULTI2VEC_CLIP, // Uncomment for testNearImage
+      Container.CONTEXTIONARY);
+  @ClassRule // Bind containers to the lifetime of the test
+  public static final TestRule _rule = compose.asTestRule();
+  private static final WeaviateClient client = compose.getClient();
+
+  private static final String COLLECTION = unique("Things");
+  private static final String VECTOR_INDEX = "bring_your_own";
+  private static final List<String> CATEGORIES = List.of("red", "green");
+
+  /**
+   * One of the inserted vectors which will be used as target vector for search.
+   */
+  private static Float[] searchVector;
+
+  @BeforeClass
+  public static void beforeAll() throws IOException {
+    createTestCollection();
+    var created = populateTest(10);
+    searchVector = created.values().iterator().next();
+  }
+
+  @Test
+  public void testNearVector() {
+    // TODO: test that we return the results in the expected order
+    // Because re-ranking should work correctly
+    var things = client.collections.use(COLLECTION);
+    var result = things.query.nearVector(searchVector,
+        opt -> opt
+            .distance(2f)
+            .limit(3)
+            .returnMetadata(MetadataField.DISTANCE));
+
+    Assertions.assertThat(result.objects).hasSize(3);
+    float maxDistance = Collections.max(result.objects,
+        Comparator.comparing(obj -> obj.metadata.distance())).metadata.distance();
+    Assertions.assertThat(maxDistance).isLessThanOrEqualTo(2f);
+  }
+
+  @Test
+  public void testNearVector_groupBy() {
+    // TODO: test that we return the results in the expected order
+    // Because re-ranking should work correctly
```

Comment:
Is this TODO still relevant? IMO, I think we should attempt to reduce, as much as possible, the amount of server behaviour testing in the clients so adding this testing wouldn't be necessary. wdyt?

Reply:
note that there is a dummy reranker module that just returns the results sorted by length - this makes it very easy to test things like the reranker

Reply:
there is also a dummy module for generative searches

Reply:
> IMO, I think we should attempt to reduce, as much as possible, the amount of server behaviour testing in the clients 

Agreed. What I meant there was that "let's somehow test that the client does not mess up the order during deserialization". I'm still thinking about how best to do that without relying on the server logic too much. Perhaps that dummy reranker would be a good solution?

Reply:
I've decided to give this a pass for now, as there isn't a way to configure a reranker on a collection. I'll drop the TODO comment, because it's not relevant to this test and will come back to that later.